### ISSUE TITLE
unpack encoding inconsistent with MRI

### ISSUE THREAD

**Issue by ankane** [jruby/jruby#5771]
### Environment

```
jruby 9.2.7.0 (2.5.3) 2019-04-09 8a269e3 Java HotSpot(TM) 64-Bit Server VM 25.192-b12 on 1.8.0_192-b12 +jit [darwin-x86_64]
```

### Expected Behavior

```ruby
p "str".unpack("H*")[0].encoding
```

returns `Encoding:US-ASCII`.

### Actual Behavior

It returns `Encoding:ASCII-8BIT`.

Edit: Narrowed down the issue to `unpack` instead of `securerandom`.

**Comment by headius** [comment]
Nice...reproduced here.

**Comment by enebo** [comment]
Too lazy to fix commit message '[master 69f6dab0a8] Fix 5771. unpack encoding inconsistent with MRI'

**Comment by ankane** [comment]
That was fast, thank you both.
